### PR TITLE
Adding cache tests

### PR DESCRIPTION
**Pull request by timkral**
@thespags 

Adding cache tests.

### CODE REVIEW

File: client/src/main/java/net/spals/oembed4j/client/cache/OEmbedResponseCache.java [hunk 1]
```diff
@@ -1,78 +0,0 @@
...
-import java.util.Optional;
-import java.util.concurrent.TimeUnit;
-
-/**
- * A cache to store {@link OEmbedResponse}s based
- * on their cache_age property.
- *
- * @author tkral
- */
-public class OEmbedResponseCache {
```
Comment (timkral) [inline client/src/main/java/net/spals/oembed4j/client/cache/OEmbedResponseCache.java:22]: Removing the cache class entirely. The logic was too complex as a separate class.

File: model/src/main/java/net/spals/oembed4j/model/OEmbedResponse.java [hunk 2]
```diff
@@ -25,6 +25,11 @@
 @JacksonXmlRootElement(localName = "oembed")
 public interface OEmbedResponse {
 
+    /**
+     * As per the oEmbed spec, version should always be 1.0
+     */
+    String DEFAULT_VERSION = "1.0";
```
Comment (timkral) [inline model/src/main/java/net/spals/oembed4j/model/OEmbedResponse.java]: Moving this so an implementing class can use it without the builder.

Comment (thespags) [reply model/src/main/java/net/spals/oembed4j/model/OEmbedResponse.java]: I've read its best practice is not to place constants in interfaces.

http://stackoverflow.com/questions/320588/interfaces-with-static-fields-in-java-for-sharing-constants

File: client/src/main/java/net/spals/oembed4j/client/JerseyOEmbedClient.java [hunk 3]
```diff
@@ -82,7 +73,7 @@ public void checkServerTrusted(final X509Certificate[] x509Certificates, final S
 
     public static JerseyOEmbedClient create(final OEmbedRegistry registry) {
         try {
-            // Configure the Jersey client
+            // 1. Configure the Jersey client
             // Setup SSL management
             // TODO: DO we need some real SSL management?
```
Comment (thespags) [inline client/src/main/java/net/spals/oembed4j/client/JerseyOEmbedClient.java:78]: Don't leave todo's

**Review by thespags — CHANGES_REQUESTED** [review]

File: model/src/main/java/net/spals/oembed4j/model/OEmbedResponse.java [hunk 4]
```diff
@@ -86,7 +86,7 @@ default boolean hasCustomProperties() {
         /**
          * As per the oEmbed spec, version should always be 1.0
          */
-        private static final String DEFAULT_VERSION = "1.0";
+        public static final String DEFAULT_VERSION = "1.0";
```
Comment (timkral) [inline model/src/main/java/net/spals/oembed4j/model/OEmbedResponse.java:89]: Constant moved back and made public.

File: client/src/main/java/net/spals/oembed4j/client/JerseyOEmbedClient.java (hunk 3 above)
Comment (timkral) [reply client/src/main/java/net/spals/oembed4j/client/JerseyOEmbedClient.java:78]: That rule assumes we have a bug tracking system setup, which we do not.

Comment (timkral) [reply client/src/main/java/net/spals/oembed4j/client/JerseyOEmbedClient.java:78]: Logged: https://github.com/spals/oembed4j/issues/7

FWIW, it was also on my notepad.

**Pull request merged**